pkg/proxy/iptables/proxier.go

```diff
@@ -1609,11 +1618,27 @@ func (proxier *Proxier) syncProxyRules() {
 
 	// Finally, tail-call to the nodeports chain.  This needs to be after all
 	// other service portal rules.
-	writeLine(proxier.natRules,
-		"-A", string(kubeServicesChain),
-		"-m", "comment", "--comment", `"kubernetes service nodeports; NOTE: this must be the last rule in this chain"`,
-		"-m", "addrtype", "--dst-type", "LOCAL",
-		"-j", string(kubeNodePortsChain))
+	addresses, err := utilproxy.GetNodeAddresss(proxier.nodePortAddresses, proxier.networkInterface)
+	if err != nil {
+		glog.Errorf("Failed to get node ip address matching nodeport cidr")
+	} else {
+		addressList := addresses.UnsortedList()
+		if len(addressList) == 1 && utilproxy.IsZeroCIDR(addressList[0]) {
+			writeLine(proxier.natRules,
```

Comment:
Can we structure this like:

cmd := []string {
    ...
}
if len(addressList) && ... {
    cmd = append(cmd, ...)
} else {
    cmd = append(cmd, ...)
}
cmd = append(cmd, ...)
writeLine(...)

So that we don't have quite so much code duplication?

Reply:
Done